### PR TITLE
feat: in aggregate rank drivers keep chunks of matches

### PR DESCRIPTION
**Pull request by JoanFM**
**Changes introduced**
Enable the ability to keep the chunks of the matches with the original retrieval score or distance.

Relates to #1491

### CODE REVIEW

File: jina/drivers/rank/aggregate/__init__.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+if False:
+    from ....types.sets import DocumentSet
+
+
+class BaseAggregateMatchesRanker(BaseRankDriver):
+    """Drivers inherited from this Driver will focus on aggregating scores from `chunks` to its `parents`."""
+
+    def __init__(self,
+                 keep_old_matches_as_chunks: bool = False,
```
Comment (nan-wang) [inline jina/drivers/rank/aggregate/__init__.py:18]:
```suggestion
                 keep_source_matches_as_chunks: bool = False,
```

File: jina/drivers/rank/aggregate/__init__.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                    |- matches: {granularity: k}
+        Output:
+        document: {granularity: k-1}
+            |- chunks: {granularity: k}
+            |    |- matches: {granularity: k}
+            |
+            |- chunks: {granularity: k}
+            |    |- matches: {granularity: k}
+            |
+            |-matches: {granularity: k-1} (Ranked according to Ranker Executor)
```
Comment (nan-wang) [inline jina/drivers/rank/aggregate/__init__.py:86]:
```suggestion
        document: {granularity: k-1}
                |- chunks: {granularity: k}
                |    |- matches: {granularity: k}
                |
                |- chunks: {granularity: k}
                    |- matches: {granularity: k}
        Output:
        document: {granularity: k-1}
                |- chunks: {granularity: k}
                |    |- matches: {granularity: k}
                |
                |- chunks: {granularity: k}
                |    |- matches: {granularity: k}
                |
                |-matches: {granularity: k-1} (Ranked according to Ranker Executor)
```

File: jina/drivers/rank/aggregate/__init__.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+class CollectMatches2DocRankDriver(BaseAggregateMatchesRanker):
+    """This Driver is intended to take a `document` with matches at a `given granularity > 0`, clear those matches and substitute
+    these matches by the documents at a lower granularity level.
+    Input-Output ::
+        Input:
+        query document: {granularity: k}
+            |- matches: {granularity: k}
+        Output:
+        query document: {granularity: k}
+            |- matches: {granularity: k-1} (Sorted according to Ranker Executor)
```
Comment (nan-wang) [inline jina/drivers/rank/aggregate/__init__.py:146]:
```suggestion
        Input:
        document: {granularity: k}
            |- matches: {granularity: k}
            
        Output:
        document: {granularity: k}
            |- matches: {granularity: k-1} (Sorted according to Ranker Executor)
```

File: jina/drivers/rank/aggregate/__init__.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+            docs_scores = self.exec_fn(match_idx, query_meta, match_meta)
+
+            self._insert_query_matches(query=context_doc,
+                                       parent_id_chunk_id_map=parent_id_chunk_id_map,
+                                       chunk_matches_by_id=matches_by_id,
+                                       docs_scores=docs_scores)
+
+
+class CollectMatches2DocRankDriver(BaseAggregateMatchesRanker):
```
Comment (nan-wang) [inline jina/drivers/rank/aggregate/__init__.py:137]: Not sure about whether it is a better naming. 
```suggestion
class AggregateMatches2DocRankDriver(BaseAggregateMatchesRanker):
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Comment by JoanFM** [comment]
Attention, changes to be added in #1337

**Comment by hanxiao** [comment]
looks like a conflict in files

**Comment by JoanFM** [comment]
> looks like a conflict in files

Done!

**Review by hanxiao — APPROVED** [review]

**Pull request merged**